libraries/core/src/main/java/com/paritytrading/philadelphia/FIXMessage.java

```diff
@@ -146,7 +159,19 @@ public FIXValue valueOf(int tag) {
      *   specified tag
      */
     public int indexOf(int tag) {
-        for (int i = 0; i < count; i++) {
+        return indexOf(tag, 0);
+    }
+
+    /**
+     * Get the index of the first instance of a field with the specified tag, starting at the specified index.
+     *
+     * @param tag the tag
+     * @param fromIndex the index to start the search from.
+     * @return the index or -1 if there are no instances of a field with the
+     *   specified tag
+     */
+    public int indexOf(int tag, int fromIndex) {
+        for (int i = fromIndex; i < count; i++) {
```

Comment:
In the `#indexOf(int,int)` method, we could simplify the resulting bytecode slightly by eliminating `i` and using `fromIndex` as the index variable within the loop. As in:
```java
for (; fromIndex < count; fromIndex++) {
    ...
```